ra2ce/graph/origins_destinations.py

```diff
@@ -455,93 +545,92 @@ def get_node_id_from_position(
-                # and if the current location is a destination (od_data[-1] is not NaN)
-                node_info["category"] = od.iloc[i][category]
-
-            # Add the new node to the graph
-            graph.add_node(new_node_id, **node_info)
-
-            # Update the inverse_nodes_dict with the new node
-            inverse_nodes_dict[match_od.coords[0]] = new_node_id
-
-            # Delete the new node from the inverse_vertices_dict as no end-points are included here
-            del inverse_vertices_dict[match_od.coords[0]]
-
-            # Update the graph edges and the inverse_vertices_dict
-            graph, inverse_vertices_dict = update_edges_with_new_node(
-                graph,
-                match_edge,
-                closest_u_v_k[0],
-                closest_u_v_k[1],
-                closest_u_v_k[2],
-                line1,
-                line2,
-                new_node_id,
-                crs,
-                inverse_vertices_dict,
-            )
-
-        except (KeyError, AssertionError):
-            # If the vertex is at the end of the road it won't be found in the inverse_vertices_dict,
-            # so search in the inverse_nodes_dict.
-            match_node = inverse_nodes_dict[
-                (closest_node_on_road[0], closest_node_on_road[1])
-            ]
-
-            # Update the node with the OD attribute
-            graph = add_data_to_existing_node(graph, match_node, match_name)
-
-            if category and od_data[-1] == od_data[-1]:
-                # If the user wants to calculate the routes to multiple locations with categories
-                # and if the current location is a destination (od_data[-1] is not NaN)
-                graph.nodes[match_node]["category"] = od.iloc[i][category]
-
-        # Save both in lists
-        od_list.append(match_od)  # save the point as a Shapely Point
+        for closest_node_on_road in closest_nodes_on_road:
+            match_od_point = Point(closest_node_on_road)
+            # Find the road to which this vertex belongs. If the vertex is on an end-point of a road, it cannot be found
+            # and it goes to the except statement.
+            try:
+                closest_u_v_k = inverse_vertices_dict[
+                    (closest_node_on_road[0], closest_node_on_road[1])
+                ]
+                match_edge = graph.edges[closest_u_v_k]
+
+                # Add the node to the graph edge
+                match_geom = match_edge["geometry"]
+
+                new_lines = split_line_with_points(match_geom, [match_od_point])
+                if len(new_lines) == 1:
+                    inverse_nodes_dict = find_closest_node(
+                        closest_node_on_road,
+                        inverse_vertices_dict,
+                        inverse_nodes_dict,
+                        graph,
+                    )
+                    assert len(new_lines) == 2
```

Comment:
This is not going to happen at this point (you entered the `if` condition with `len(new_lines) == 1`.

Reply:
Remove this line.